### PR TITLE
feat: Add snowflake integration

### PR DESCRIPTION
**Pull request by rajeshneti**
**Fixes** # (*issue*)
Notion link: https://www.notion.so/rudderstacks/Terraform-Snowflake-Integration-3cf76c8d989346eba74b2218dc8e19c9
## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update
## Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have added unit tests for the code
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: rudderstack/integrations/destinations/destination_snowflake_test.go [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+				},
+				"useRudderStorage": false,
+				"additionalProperties": true,
+				"jsonPaths": "./example-paths",
+				"cloudProvider": "AWS",
+				"prefix": "example-prefix",
+        "bucketName": "example-bucket-name",
+        "accessKeyID": "example-access-key-id",
+        "accessKey": "example-access-key",
+        "enableSSE": true
```
Comment (ktgowtham) [inline rudderstack/integrations/destinations/destination_snowflake_test.go:78]: Can we fix the formatting

Comment (ktgowtham) [reply rudderstack/integrations/destinations/destination_snowflake_test.go:78]: let's add more test cases? for other storage providers?

Comment (rajeshneti) [reply rudderstack/integrations/destinations/destination_snowflake_test.go:78]: yeah, I have added for all the three providers AWS, GCP and azure

File: docs/resources/destination_snowflake.md [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+- `created_at` (String) Time when the resource was created, in ISO 8601 format.
+- `id` (String) The ID of this resource.
+- `updated_at` (String) Time when the resource was last updated, in ISO 8601 format.
+
+<a id="nestedblock--config"></a>
+### Nested Schema for `config`
+
+Required:
+
+- `account` (String) Account ID of your Snowflake warehouse. This account ID is part of the Snowflake URL.
```
Comment (ameypv-rudder) [inline docs/resources/destination_snowflake.md:80]: Can we add the FAQ describing the various Snowflake account ID examples here?

https://www.rudderstack.com/docs/destinations/warehouse-destinations/faq/#while-configuring-the-snowflake-destination-what-should-i-enter-in-the-account-field

**Review by ameypv-rudder — APPROVED** [review]
LGTM! Just one minor suggestion that can be incorporated.

File: docs/resources/destination_snowflake.md (hunk 2 above)
Comment (rajeshneti) [reply docs/resources/destination_snowflake.md:80]: This is an auto generated doc generated from the go file. I couldn't find a way to add the example.

Comment (ameypv-rudder) [reply docs/resources/destination_snowflake.md:80]: I think that should be fine!

Comment (rajeshneti) [reply docs/resources/destination_snowflake.md]: done

**Review by ktgowtham — APPROVED** [review]

**Review by fxenik — APPROVED** [review]

**Pull request merged**